src/axom/core/ArrayView.hpp

```diff
@@ -146,7 +146,11 @@ class ArrayView : public ArrayBase<T, DIM, ArrayView<T, DIM, SPACE>>
     assert(offset + count <= m_num_elements);
     ArrayView slice = *this;
     slice.m_data += offset;
-    if(count != -1)
+    if(count == -1)
```

Comment:
Suggestion:
I think we should enforce/assert  that offset is in range, i.e. `offset >= 0 && offset < m_num_elements`
Also, instead of `count == -1` perhaps it should check for `count < 0 `.

Negative indices don't show up in `std::span::subspan` since they use an unsigned int (`std::size_t`) -- https://en.cppreference.com/w/cpp/container/span/subspan

The practical difference is what would happen if someone called `subspan(n, -2)`

Reply:
Great suggestion.  Will do.